packages/core/src/semantic-icon-provider/SemanticIconProvider.tsx

```diff
@@ -0,0 +1,58 @@
+import {
+  ChevronDownIcon,
+  ChevronRightIcon,
+  ChevronUpIcon,
+  ErrorSolidIcon,
+  InfoSolidIcon,
+  OverflowMenuIcon,
+  SuccessTickIcon,
+  UserSolidIcon,
+  WarningSolidIcon,
+} from "@salt-ds/icons";
+import type React from "react";
+import { type FC, type ReactNode, createContext, useContext } from "react";
+
+type IconMap = {
+  SemanticChevronDownIcon?: React.ElementType;
+  SemanticChevronUpIcon?: React.ElementType;
+  SemanticErrorSolidIcon?: React.ElementType;
+  SemanticSuccessTickIcon?: React.ElementType;
+  SemanticInfoSolidIcon?: React.ElementType;
+  SemanticWarningSolidIcon?: React.ElementType;
+  SemanticChevronRightIcon?: React.ElementType;
+  SemanticOverflowMenuIcon?: React.ElementType;
+  SemanticUserSolidIcon?: React.ElementType;
+};
+
+interface SemanticIconProviderProps {
+  iconMap?: IconMap;
+  children: ReactNode;
+}
+
+const defaultIconMap: IconMap = {
+  SemanticChevronDownIcon: ChevronDownIcon,
+  SemanticChevronUpIcon: ChevronUpIcon,
+  SemanticErrorSolidIcon: ErrorSolidIcon,
+  SemanticSuccessTickIcon: SuccessTickIcon,
+  SemanticInfoSolidIcon: InfoSolidIcon,
+  SemanticWarningSolidIcon: WarningSolidIcon,
+  SemanticChevronRightIcon: ChevronRightIcon,
+  SemanticOverflowMenuIcon: OverflowMenuIcon,
+  SemanticUserSolidIcon: UserSolidIcon,
+};
+
+const SemanticIconContext = createContext<IconMap>(defaultIconMap);
+
+export const SemanticIconProvider: FC<SemanticIconProviderProps> = ({
+  iconMap = defaultIconMap,
+  children,
+}) => (
+  <SemanticIconContext.Provider value={iconMap}>
+    {children}
+  </SemanticIconContext.Provider>
+);
+
+export const useSemanticIcon = () => {
+  const context = useContext(SemanticIconContext);
+  return context || defaultIconMap;
```

Comment:
As best practice should we add a try/catch that logs if you have used the hook without the provider ?

Reply:
thinking about it, would be a breaking change without a try/catch?

Reply:
we have kept it like intentionally, if someone, won't use provider, then the default icons map will work

Reply:
it will throw a nasty error though ?

Reply:
haven't seen any error in console, without provider

Reply:
What if a custom partial map is provided?